view/template/layout/_analytics_footer.php

```diff
@@ -8,13 +8,16 @@
 
     ga('create', 'UA-60403362-1', 'auto');
     ga('require', 'GTM-NT8579P');
+    ga('set', 'userId', '<?php echo $_SESSION[Session::USER_ID] ?>');
     ga('send', 'pageview');
 
    !function(f,b,e,v,n,t,s) {if(f.fbq)return;n=f.fbq=function(){n.callMethod? n.callMethod.apply(n,arguments):n.queue.push(arguments)}; if(!f._fbq)f._fbq=n;n.push=n;n.loaded=!0;n.version='2.0'; n.queue=[];t=b.createElement(e);t.async=!0; t.src=v;s=b.getElementsByTagName(e)[0]; s.parentNode.insertBefore(t,s)}(window,document,'script', 'https://connect.facebook.net/en_US/fbevents.js');
 
-   fbq('init', '1618717031725766');
+   fbq('init', '1618717031725766',{uid: '<?php echo $_SESSION[Session::USER_ID] ?>'}));
    fbq('track', '<?php echo Response::getFacebookPixelAnalyticsType() ?>');
 
+    <?php LBRY::logWebVisitor('lbry.io', $_SESSION[Session::USER_ID], $_SERVER['REMOTE_ADDR'])?>
```

Comment:
1) This breaks MVC.
2) Isn't this redundant if it is in `Session::init()`? (Session::init is called on every execution, I believe)

Reply:
You are right, init is called on every execution. I removed that line since it is already in the init.

Reply:
it appears to get called twice for me...2 seconds apart.

Reply:
It's going to get called once on init (happens early in execution) as well as when this footer is rendered (towards the end of execution). I hope it doesn't take a full 2s to render a page though!

Reply:
I removed it from the footer. So its somewhere else.